src/sqliteformat.cpp

```diff
@@ -391,6 +392,17 @@ bool SqliteFormat::modifyComponents(const Incidence::Ptr &incidence, const QStri
         }
     }
 
+    if (oldNotebook != nbook && dbop == DBUpdate) {
+        qCDebug(lcMkcal) << "changing notebook from" << oldNotebook << "to" << nbook;
+        if (!modifyComponents(incidence, nbook, DBMarkDeleted)) {
+            qCWarning(lcMkcal) << "failed to mark incidence" << incidence->uid()
+                               << incidence->recurrenceId() << "as deleted for notebook change";
+            goto error;
+        } else {
+            dbop = DBInsert;
```

Comment:
So guess this effectively creates a new row with a new component id. Quite special so could add some note.

But how about if the instance is moved from notebook a->b and then back b->a. The latter should update the deleted instance as undeleted?

Reply:
Eh eh, good point. This will fail at the moment. I'll need to move the purge delete action on addition from the `SqliteStorage::Privae::saveIncidence()` to here, to ensure that any addition don't leave any mark as deleted entries.